### DISCUSSION TITLE
Creating a Vulkan window

### DISCUSSION THREAD

**Discussion by isolin** [opentk/opentk#1751] [Q&A] [answered]
I tried to run the [Vulkan](https://github.com/opentk/opentk/blob/opentk5.0/tests/VulkanTestProject/Program.cs) example, but it is failing on

```cs
// FIXME: How do we create a window for OpenGL vs Vulkan?
Window = Toolkit.Window.Create(null);
```

Are there any updates on this? It seems that a Vulkan specific class that would inherit from `GraphicsApiHints` is missing so far.

Moreover, `Toolkit.Vulkan` is `null` a few lines later. Not sure if the window initialization should handle that or whether it is independent.

**Comment by NogginBops** [comment]
I think the issue here is that I've unfortunately not had the time to implement a vulkan component for Linux. Implementing it shouldn't be that hard, I'll see if I can get to it pretty soon.

Making a new preview should be quite easy, but I'd like to also fix some of the other issues that have been reported for the next preview. But if that takes too long to do then I'll just release a new preview that just that.

Fortunately vulkan doesn't need a vulkan component , you can just use the vulkan API directly. Unfortunately in the X11 backend we currently crash if you pass null when creating a window. Maybe you could try making the window with opengl settings but never create the context and instead use vulkan. Not sure if that would work...

**Reply by NogginBops** [reply, marked as answer]
Ok I spent some time to create a X11VulkanComponent and get a working window.
So as soon as https://github.com/opentk/opentk/pull/1742 is merged and released vulkan support will work on linux.
![image](https://github.com/user-attachments/assets/2b48903f-c6af-48c6-8bf6-6efb330cc244)

**Comment by isolin** [comment]
Thank you very much! Looking forward for pre12 :)

So if I comment out all the lines with `Toolkit.Window` it should still work headless? As mentioned above, I am getting a `NullPointerException` then, on
```cs
var requiredExtensions = Toolkit.Vulkan.GetRequiredInstanceExtensions();
```
since `Toolkit.Vulkan` is `null`. And out of OpenTK Vulkan is running fine on my machine, so there should be no installation or driver issues.

```cs
var _vulkanComponent = PlatformComponents.CreateVulkanComponent();
```
throws `An unhandled exception of type 'System.NotSupportedException' occurred in OpenTK.Platform.dll: 'This platform doesn't support the OpenTK.Platform.IVulkanComponent component yet.'`

Ah, now I see that you just implemented the component in the new PR. Ok, so I'll wait for pre12 :)

**Reply by NogginBops** [reply]
I'll try and get pre12 out as soon as I'm able :)

**Reply by NogginBops** [reply]
I've just released 5.0-pre.12!
https://www.nuget.org/packages/OpenTK/5.0.0-pre.12

This version implements `Toolkit.Vulkan` on linux, so it should work for you. @isolin

**Reply by isolin** [reply]
Yes this works!
